poppy/wfe.py

```diff
@@ -241,14 +254,26 @@ def get_opd(self, wave):
                     noll_normalize=True
                 )
             else:
-                combined_zernikes += k_in_m * zernike.cached_zernike1(
-                    j,
-                    wave.shape,
-                    pixelscale_m,
-                    self.radius.to(u.meter).value,
-                    outside=0.0,
-                    noll_normalize=True
-                )
+                try:
+                    combined_zernikes += k_in_m * zernike.cached_zernike1(
+                        j,
+                        wave.shape,
+                        pixelscale_m,
+                        self.radius.to(u.meter).value,
+                        outside=0.0,
+                        noll_normalize=True
+                    )
+                except: # can't use cached_zernikes if the user changed between CPU and CuPy
```

Comment:
Better to have a specific exception that is caught here, like `except MathError:` or whatever the specific exception is. It's recommended Python style to not have bare `except`.